src/Bindings/ManualBindings.cpp

```diff
@@ -110,6 +111,146 @@ static int tolua_Clamp(lua_State * tolua_S)
 
 
 
+static int tolua_CompressStringZLIB(lua_State * tolua_S)
+{
+	cLuaState S(tolua_S);
+	if (
+		!S.CheckParamString(1) &&
+		(
+			!S.CheckParamNumber(2) ||
```

Comment:
Nope, this should be an AND, and all the others should be ORs. You want to fail if any of the top-level tests fail.